EIPS/eip-6492.md

```diff
@@ -0,0 +1,143 @@
+- If all this fails, try `ecrecover` verification
+
+## Rationale
+
+We believe that wrapping the signature in a way that allows to pass the deploy data is the only clean way to implement this, as it's completely contract agnostic, but also easy to verify.
+
+The wrapper format ends in `magicBytes`, which ends with a `0x69`, which makes it is impossible for it to collide with a valid `ecrecover` signature if packed in the `r,s,v` format, as `0x69` is not a valid value for `v`.
+
+The verification order to ensure correct verification is based on the following rules:
+
+- checking `ecrecover` MUST NOT happen before [ERC-1271](./eip-1271.md) verification, because a contract may use a signature format that also happens to be a valid `ecrecover` signature for an EOA with a different address. One such example is a contract that's a wallet controlled by said EOA.
+- checking for `magicBytes` MUST happen before `ecrecover` in order to avoid trying to verify a counterfactual contract signature via `ecrecover` if such is clearly identifiable
+- checking for contract code MUST happen before checking for `magicBytes`, to avoid possible collisions, as [ERC-1271](./eip-1271.md) allows any arbitrary values for signatures
+
+## Backwards Compatibility
+
+This ERC is backward compatible with previous work on signature validation, including [ERC-1271](./eip-1271.md) and allows for easy verification of all signature types, including EOA signatures and typed data ([EIP-712](./eip-712.md)). 
+
+## Reference Implementation
+
+Example implementation of a verification contract:
+
+```solidity
+contract ERC6492Verifier {
+  /**
+   * @notice Verifies that the signature is valid for that signer and hash
+   */  
+  function isValidUniversalSignature(
+    address _signer,
+    bytes32 _hash,
+    bytes calldata _signature
+  ) external override returns (bool) {
+    uint32 contractSize;
+    assembly {
+      size := extcodesize(_signer)
+    }
+    if (contractSize > 0) {
+      return IERC1271Wallet(_signer).isValidSignature(_hash, _signature) == 0x1626ba7e;
+    }
+    // NOTE: bytes library not included
+    if (_signature.readBytes4(_signature.length - 4) == 0x69696969) {
+      _signature.trimToSize(_signature.length - 4);
+      address create2Factory;
+      bytes memory factoryCalldata;
+      (create2Factory, factoryCalldata, sig) = abi.decode(sig, (address, bytes, bytes));
+      (bool success, ) = create2Factory.call(factoryCalldata);
+      if (!success) return false;
+      return IERC1271Wallet(_signer).isValidSignature(_hash, sig) == 0x1626ba7e;
+    }
+    
+    // ecrecover verification
+    require(_signature.length == 65, "SignatureValidator#recoverSigner: invalid signature length");
+    uint8 v = uint8(_signature[64]);
+    bytes32 r = _signature.readBytes32(0);
```

Comment:
If you want to reduce your dependency on external libraries, you could use bytes32(signature[0:32])
If you don't care about reducing library dependencies and would like to shorten the sample code then you could use OpenZeppelin.ECDSA library (or OpenZeppelin.ERC1271)

Reply:
thanks, IMO the example should have as few dependencies as possible